tango/integrations/torch/train_callback.py

```diff
@@ -21,14 +21,18 @@ class TrainCallback(Registrable):
         All of the parameters to this base class will be automatically set within
         the training loop, so you shouldn't include them in your config for your callbacks.
 
+    .. important::
+        The ``step`` argument to callback methods is the total/overall number of training steps
+        so far, independent of the current epoch.
+
     .. seealso::
         See :class:`~tango.integrations.wandb.WandbTrainCallback` for an example
         implementation.
 
     Attributes
     ----------
     train_config : :class:`TrainConfig`
-    model : :class:`Model`
+    training_engine : :class:`TrainingEngine`
```

Comment:
Is that change necessary? If `TrainCallback` becomes a major integration point, it will be easier to understand how to write one if you get a `Model` instead of a `TrainingEngine`. But that's only true if those callbacks don't need access to the engine.

Reply:
Well, a `TrainingEngine` is responsible for constructing the `Model` (since it might have its own, more memory efficient way of doing so), so I think it makes sense for the `TrainingEngine` to own the model. That said, I see your point. It's important that each `TrainCallback`s can access the model. And they can, since part of the `TrainingEngine` API is that the model is accessible through [`TrainingEngine.model`](https://github.com/allenai/tango/blob/ee179adff1927c1cc0aaae84db9b23b2759b6a06/tango/integrations/torch/training_engine.py#L26). But this should be more obvious. Here's my solution to that: 12294d7

Reply:
Can you add a note to the docs for `TrainCallback` that this is how you get to the model? I can imagine an impatient coder looking at the docs for `TrainCallback` and not finding it. I can imagine that because I have been that coder.

Reply:
Done: 62a8ad9